packages/components/phone-number/src/components/osds-phone-number/osds-phone-number.tsx

```diff
@@ -1,33 +1,98 @@
-import { Component, Host, h, Prop } from '@stencil/core';
+import type { OdsPhoneNumberAttribute } from './interfaces/attributes';
+import { ODS_COUNTRY_ISO_CODE, ODS_LOCALE } from '@ovhcloud/ods-common-core';
+import { Component, Host, h, Prop, State, Watch } from '@stencil/core';
 import { ODS_INPUT_TYPE } from '@ovhcloud/ods-component-input';
 import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
-import { OdsPhoneNumberAttribute } from './interfaces/attributes';
+import { OdsPhoneNumberController } from './core/controller';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { ODS_PHONE_NUMBER_COUTRIE } from './constants/phone-number-countries';
 
 /**
  * @slot (unnamed) - Phone Number content
  */
 @Component({
   tag: 'osds-phone-number',
   styleUrl: 'osds-phone-number.scss',
-  shadow: true
+  shadow: true,
+  assetsDirs: ['../../assets'],
 })
 export class OsdsPhoneNumber implements OdsPhoneNumberAttribute {
+  controller = new OdsPhoneNumberController(this);
+
   /** @see OdsPhoneNumberAttribute.clearable */
   @Prop({ reflect: true }) clearable?: boolean = DEFAULT_ATTRIBUTE.clearable;
 
+  /** @see OdsPhoneNumberAttribute.countries */
+  @Prop({ reflect: true, mutable: true }) countries?: ODS_COUNTRY_ISO_CODE[] | ODS_PHONE_NUMBER_COUTRIE.All = DEFAULT_ATTRIBUTE.countries;
+
   /** @see OdsPhoneNumberAttribute.disabled */
   @Prop({ reflect: true }) disabled?: boolean = DEFAULT_ATTRIBUTE.disabled;
 
   /** @see OdsPhoneNumberAttribute.error */
   @Prop({ reflect: true }) error?: boolean = DEFAULT_ATTRIBUTE.error;
 
+  /** @see OdsPhoneNumberAttribute.isoCode */
+  @Prop({ reflect: true, mutable: true }) isoCode?: ODS_COUNTRY_ISO_CODE = DEFAULT_ATTRIBUTE.isoCode;
+
+  /** @see OdsPhoneNumberAttribute.locale */
+  @Prop({ reflect: true, mutable: true }) locale?: ODS_LOCALE = DEFAULT_ATTRIBUTE.locale;
+
   /** @see OdsPhoneNumberAttribute.value */
   @Prop({ reflect: true, mutable: true }) value = DEFAULT_ATTRIBUTE.value;
 
+  @State() i18nCountriesMap!: Map<string, { isoCode: string , name: string }>;
+
+  @State() countriesList: readonly ODS_COUNTRY_ISO_CODE[] = [];
+  @State() hasCountries: boolean = false;
+
+  componentWillLoad(): void {
+    this.isoCode = this.controller.getDefaultIsoCode();
+    this.locale = this.controller.getDefaultLocale();
+    this.handlerLocale(this.locale);
+    this.handlerCountries();
+  }
+
+  @Watch('locale')
+  handlerLocale(locale: ODS_LOCALE): void {
+    const translationFile = this.controller.loadTranslationFileByLocale(locale);
+    this.i18nCountriesMap = translationFile.reduce((acc, country) => {
+      acc.set(country.isoCode, country);
+      return acc;
+    }, new Map());
+  }
+
+  @Watch('countries')
+  handlerCountries(): void {
+    this.hasCountries = Boolean(this.countries?.length);
```

Comment:
did we go with Boolean instead of `!!`, not sure anymore 😅

Reply:
We say both but I can replace the Boolean